File: html/cgi-bin/cookie.py
#!/usr/bin/env python3

import cgi
import cgitb
import http.cookies

cgitb.enable()

# Get the CGI data
form = cgi.FieldStorage()

# Create a new cookie
cookie = http.cookies.SimpleCookie()

# Check if the 'name' parameter is provided
username = form.getfirst('username', '')
if username:
    # Set the cookie value
    cookie['username'] = username

    # Set cookie expiration time (in seconds)
    cookie['username']['expires'] = 3600  # Cookie expires in 1 hour

# Set the Content-Type header to 'text/html'
print("Content-Type: text/html")
print(cookie)
print()

# Print the HTML content
print("<html>")
print("<head>")
print("<title>Cookie Example</title>")
print("</head>")
print("<body>")
print("<h1>Cookie Example</h1>")

# Print the cookie value if it exists
if 'username' in cookie:
    print("<p>Welcome back, {}!</p>".format(username))
else:
    print("<p>Please enter your name:</p>")
    print('<form method="post" action="">')
    print('<input type="text" name="name" required>')
    print('<input type="submit" value="Submit">')
    print('</form>')

print("</body>")
print("</html>")
print()
print()

def set_cookie(username):
	cookie = http.cookies.SimpleCookie()
	cookie['username'] = username
	if expires:
		cookie[name]['expires'] = expires
	print(cookie)

File: html/cgi-bin/cookie.php
<?php phpinfo(); ?>
<?php
// Start the session and register the session variables
session_start();

// Check if the username is received from the GET request
if (isset($_GET['username'])) {
    // Sanitize the username input to prevent security issues
    $username = filter_input(INPUT_GET, 'username', FILTER_SANITIZE_SPECIAL_CHARS);

    // Set a session variable for the username
    $_SESSION['username'] = $username;

    // Set a cookie for the username that expires in 1 hour
    setcookie('username', $username, time() + 3600, "/");

    // Display a welcome message
    echo "<!DOCTYPE html>
    <html lang='en'>
    <head>
        <meta charset='UTF-8'>
        <meta name='viewport' content='width=device-width, initial-scale=1.0'>
        <title>Welcome</title>
        <link rel='stylesheet' href='/css/style.css'>
    </head>
    <body>
        <div>
            <h1>Welcome, " . htmlspecialchars($username) . "!</h1>
            <p>You are now logged in.</p>
        </div>
    </body>
    </html>";
} 
else 
{
    // Redirect back to the form if the username is not provided
    header("Location: /index.html");
    exit;
}
?>
